lib/lib-dynamodb/README.md

```diff
@@ -0,0 +1,163 @@
+# @aws-sdk/lib-dynamodb
+
+[![NPM version](https://img.shields.io/npm/v/@aws-sdk/lib-dynamodb/latest.svg)](https://www.npmjs.com/package/@aws-sdk/lib-dynamodb)
+[![NPM downloads](https://img.shields.io/npm/dm/@aws-sdk/lib-dynamodb.svg)](https://www.npmjs.com/package/@aws-sdk/lib-dynamodb)
+
+## Overview
+
+The document client simplifies working with items in Amazon DynamoDB by
+abstracting away the notion of attribute values. This abstraction annotates native
+JavaScript types supplied as input parameters, as well as converts annotated
+response data to native JavaScript types.
+
+## Marshalling Input and Unmarshalling Response Data
+
+The document client affords developers the use of native JavaScript types
+instead of `AttributeValue`s to simplify the JavaScript development
+experience with Amazon DynamoDB. JavaScript objects passed in as parameters
+are marshalled into `AttributeValue` shapes required by Amazon DynamoDB.
+Responses from DynamoDB are unmarshalled into plain JavaScript objects
+by the `DocumentClient`. The `DocumentClient` does not accept
+`AttributeValue`s in favor of native JavaScript types.
+
+### Example
+
+Here is an example `Key` which is sent to DynamoDB client in GetItem operation:
+
+```json
+{ "KEY_NAME": { "ATTRIBUTE_VALUE": VALUE } }
+```
+
+where `ATTRIBUTE_VALUE` can be one of the following:
+
+| ATTRIBUTE_VALUE | type       | TypeScript type                        |
+| --------------- | ---------- | -------------------------------------- |
+| NULL            | Null       | null                                   |
+| BOOL            | Boolean    | boolean                                |
+| BS              | Binary Set | Set\<Unit8Array\>                      |
+| NS              | Number Set | Set\<number \| BigInt\>                |
+| SS              | String Set | Set\<string\>                          |
+| S               | String     | string                                 |
+| B               | Binary     | Uint8Array, Buffer, File, Blob...      |
+| M               | Map        | { [key: string]: \<ATTRIBUTE_VALUE\> } |
+| L               | List       | \<ATTRIBUTE_VALUE\>[]                  |
+| N               | Number     | number, BigInt                         |
+
+The DynamoDB document client should abstract the attribute values as follows in
+both input and output:
+
+```json
+{ "KEY_NAME": VALUE }
+```
+
+## Usage
+
+To create document client you need to create DynamoDB client first as follows:
+
+```js
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb"; // ES6 import
+// const { DynamoDBClient } = require("@aws-sdk/client-dynamodb"); // CommonJS import
+
+// Bare-bones DynamoDB Client
+const client = new DynamoDBClient({});
+```
+
+```js
+import { DynamoDB } from "@aws-sdk/client-dynamodb"; // ES6 import
+// const { DynamoDB } = require("@aws-sdk/client-dynamodb"); // CommonJS import
+
+// Bare-bones DynamoDB Client
```

Comment:
Isn't this one the full DynamoDB Client?

Reply:
> Isn't this one the full DynamoDB Client?

It has all the features of Document Client, but we call it `"bare-bones"` as you need to import Commands separately. The `"full"` client is the one where importing Commands is not required.

Reply:
Right, but looking at the code, `DynamoDBClient` is the bare-bones version, while `DynamoDB` has a ton of functions (matching the v2 style).

Reply:
Yea, this should be `DynamoDBClient` followed by the "Commands" to be imported. E.g.

```ts
import {
  DynamoDBClient,
  QueryCommand,
  ScanCommand,
  GetItemCommand,
  PutItemCommand,
  UpdateItemCommand,
  BatchGetItemCommand,
  BatchWriteItemCommand,
  WriteRequest,
  KeysAndAttributes,
  AttributeValueUpdate,
} from '@aws-sdk/client-dynamodb'
```

Reply:
My bad, forgot to replace while copy-pasting from bare bones client.
```suggestion
// Full DynamoDB Client
```

Reply:
Figured it was copy pasta